pkg/operator/resource/resourceread/core.go

```diff
@@ -30,7 +30,20 @@ func ReadSecretV1OrDie(objBytes []byte) *corev1.Secret {
 	if err != nil {
 		panic(err)
 	}
-	return requiredObj.(*corev1.Secret)
+
+	secret := requiredObj.(*corev1.Secret)
+	if len(secret.StringData) > 0 {
```

Comment:
This fixes one possible inbound source.  Instead of this, we should update the appySecret function to put the StringData into the Data field and then update based on that.

Reply:
The reason I did it here and not there is that you need to process/modify the `.Data` before sending it to `ApplySecret`. E.g. I need to inject certificates. If we don't do it at read time, any processing loop will need to be duplicated to handle both `.Data` and `.StringData`. Generally the API assumptions is you get only `.Data` when reading. Unifying that assumption for reading manifests seems like the closest we can get to have sane (non-duplicated) processing. As the base64 difference is meant just for writing binary data in text files I don't see a use case for setting `StringData` on any code path, hence `ApplySecret` should't need to handle it and it explicitly forbids it in case someone made a mistake.

Reply:
(the base64 conversion happens when decoding the object)

Reply:
> `ApplySecret` should't need to handle it and it explicitly forbids it in case someone made a mistake.

We should match.  Either this isn't allowed and the code is correct and people will be smacked.  Or we change the code to allow on both sides. But a halfway state isn't good for anyone.

Reply:
Half way prevents bugs like if somebody reads secrets with `StringData` without merging it into `.Data`, processes it with some function (those generally handle only `.Data`), any data stored in `.StringData` are missed in the processing but ApplySecret stops you on StringData and you discover the bug.